instrumentation/tomcat-7.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/tomcat7/TomcatTracer.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.tomcat7;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.instrumentation.api.servlet.AppServerBridge;
+import io.opentelemetry.instrumentation.api.tracer.HttpServerTracer;
+import java.net.URI;
+import java.util.Collections;
+import org.apache.coyote.Request;
+import org.apache.coyote.Response;
+import org.apache.tomcat.util.buf.MessageBytes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TomcatTracer extends HttpServerTracer<Request, Response, Request, Request>
+    implements TextMapPropagator.Getter<Request> {
+
+  private static final Logger log = LoggerFactory.getLogger(TomcatTracer.class);
+  private static final TomcatTracer TRACER = new TomcatTracer();
+
+  public static TomcatTracer tracer() {
+    return TRACER;
+  }
+
+  public Context startServerSpan(Request request, String adapterClassName) {
```

Comment:
Pass the whole `Method` here instead of `adapterClassName`. There is `getSpanName(Method)`

Reply:
But I don't have the method 🤔

Reply:
But you can have :) Like servlet advices do

Reply:
Now you lost me. Servlet advices set server span name from the URL obtained from the Request. What method are you talking about?

Reply:
Take a look at `io.opentelemetry.javaagent.instrumentation.servlet.http.HttpServletResponseInstrumentationModule.SendAdvice`

Reply:
ok